#167190519 Setup Database With Sequelize ORM

chialuka opened the pull request:
#### What does this PR do?
- Setups databases for the project. The PR creates test, development, and production databases

- Configures the project to use Sequelize ORM

- Configures Sequelize to use ES6 syntax

#### Description of Task proposed in this pull request?
- Sequelize by default uses ES5. The PR uses the .sequelize.rc file to configure Sequelize to use ES6 syntax. It also provides PostgreSQL databases for development and production and an SQLite database for testing.

#### How should this be manually tested (Quality Assurance)?
- There are no changes to the API as this is a chore that sets up the database for the project

#### What are the relevant pivotal tracker stories?
- [#167190519](https://www.pivotaltracker.com/story/show/167190519)

#### Any background context you want to add (Operations Impact)?
- Add a database url connection to the .env file for the DB_URL_DEV value to run a database for the development environment. Also, set NODE_ENV to development. 

To run migrations and generate models, `sequelize`, `sequelize-cli`, `pg`, and `pg-hstore` must be installed as dependencies.

#### What I have learned working on this feature
- I have learned how to use Sequelize, sequelize-cli, sqlite and how to use ES6 syntax with the configuration. I also learned how to set and view NODE_ENV using the terminal on a Mac.

ayodejiAA reviewed (dismissed):
Nice work

dinobi commented on .sequelizerc:
```diff
@@ -0,0 +1,11 @@
+require("@babel/register");
+require("@babel/register");
```
why the duplicate?

chialuka replied:
That's an error. It's been fixed now.

dinobi commented on .sequelizerc:
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  'config': path.resolve('server/database/config', 'config.js'),
+  'models-path': path.resolve('server/database/models'),
+  'seeders-path': path.resolve('server/database/seeders'),
```
this reads better: `'seeds-path': path.resolve('server/database/seeds'),`

chialuka replied:
Okay. Those folders are generated automatically by Sequelize. I will rename it to seeds

dinobi reviewed (commented):
Please update your branch

dinobi commented on __test__/index.test.js:
```diff
@@ -0,0 +1,31 @@
+
+import request from 'supertest';
```
Why do we still have supertest here?

chialuka committed:
chore(database):setup database with sequelize ORM
- create .sequelizerc file
- initialize sequelize-cli for database
- configure sequelize to use ES6
[Finishes #167190519]

chialuka committed:
Merge branch 'develop' into ch-setup-postgres-db-167190519
